kernel/constr.ml

```diff
@@ -744,7 +744,6 @@ let fold_map f accu c = match kind c with
       if t' == t then accu, c
       else accu, mkProj (p, t')
   | Evar (e,l) ->
-    (* Doesn't matter, we should not hashcons evars anyways *)
```

Comment:
Why remove this comment? It's about hashconsing the list, not the subterms.

Reply:
It was added in 45e14bdb854fe5da40c2ed1d9ca575ae8d435d36 which changes Array.Smart.fold_left_map to List.fold_left_map
Since we use List.Smart.fold_left_map now it doesn't make sense anymore

Reply:
Yes, I wrote that comment. The point is not that we're smart or not, the point is that the list in itself is not hashconsed.